app/components/table_component/table_component.html.haml

```diff
@@ -1,7 +1,8 @@
-%table.table-content{id: @id, class: stripped_class + ' ' + borderless_class + ' ' + layout_fixed_class + @custom_class}
-  %thead
-    = header
-  %tbody{id: "#{@id}_table_body"}
-    - rows.each do |row|
-      = row
-    = content
\ No newline at end of file
+%div{'data-controller': 'table-component', 'data-table-component-sort-value': @sort.to_s, 'data-table-component-defaultsortcolumn-value': @default_sort_column, 'data-table-component-paging-value': @paging.to_s, 'data-table-component-searching-value': @searching.to_s}
```

Comment:
Actually I added the div wrapping the table to avoid an error I get when I put the data controller directly in the table tag, when I do this, the init of the table (new table ...) will be executed many times because the data tables re-draw the table and this does a kinda recursivity